### ISSUE TITLE
Passing colorbar=True to a datashaded element could raise a warning

### ISSUE THREAD

**Issue by ahuang11** [ioam/holoviews#2873]
```
points = df.hvplot.points('index', 'value')
datashade(points).options(colorbar=True)
```

**Comment by jbednar** [comment]
Makes sense; can mostly get away with it, but it has annoying implications like this one.  @ahuang11 ahuang11 seems talented enough to try splitting the class in this way... :-)

**Comment by ahuang11** [comment]
Heh, so the issue stems from RGB inheriting from Image or am I completely mistaken?
https://github.com/ioam/holoviews/blob/0a154d72f9a6fff737b84e7cd21d875d636944b6/holoviews/element/raster.py#L565-L663

**Comment by jbednar** [comment]
No, I think RGB inheriting from Image is by design; semantically, it's a type of Image. But these are the element classes; what's shared is the bokeh and matplotlib *plotting* class RGBPlot.  Specifically, there's no separate ImagePlot class.  But we can't simply copy RGBPlot to another one ImagePlot and omit the colorbar option, because RGBPlot inherits from RasterPlot which inherits from ColorbarPlot, which is presumably what makes it support the colorbar option.  So it will take a bit of juggling classes to get things working properly...

**Comment by philippjfr** [comment]
I wasn't quite right about that. In the matplotlib backend the ``RasterPlot`` class handles Rasters, Images and RGBs. In the bokeh backend the problem is that ``RGBPlot`` inherits the ``colorbar`` parameter from ``RasterPlot``. So what would need to happen is that in both backends we need to ensure that the ``RGBPlot`` class does not inherit from ``ColorbarPlot``.
